tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -345,23 +345,32 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
 
             TraceContext traceContext;
 
-            // try to get the trace context (from local spans) or
-            // sampling priority (from propagated spans),
-            // otherwise start a new trace context
+            // try to get the trace context (from local spans),
+            // otherwise start a new trace context and get sampling priority (from propagated spans)
             if (parent is SpanContext parentSpanContext)
             {
+                // if traceContext is not null, parent is from a local (non-propagated) span
+                // and this child span belongs in the same TraceContext
                 traceContext = parentSpanContext.TraceContext;
+
                 if (traceContext == null)
                 {
+                    // if traceContext is null, parent was extracted from propagation headers.
+                    // start a new trace and keep the sampling priority and trace tags.
                     var traceTags = TagPropagation.ParseHeader(parentSpanContext.PropagatedTags);
                     traceContext = new TraceContext(this, traceTags);
```

Comment:
BTW, this is the magic sauce that will pass `_dd.p.dm` and other propagated tags from a distributed trace's `x-datadog-tags` header into the local span's `span.Context.TraceContext.Tags` collection